lib/app/utils.js

```diff
@@ -19,6 +19,67 @@ export function globalHandleError(error) {
   }
 }
 
+// Takes an array of errors and assigns a message and stack to it to make it error-like:
+export function compactMultipleErrors(errors, messagePrefix = 'Multiple Errors', errorJoiner = '\n + \n'){
+  if (!Array.isArray(errors)) {
+    throw new TypeError('Supplied errors is not an array');
+  }
+  
+  return Object.assign(errors, {
+    message: `${messagePrefix}:\n${
+      errors
+        .map(error => 
+          (typeof error === 'object' && error !== null && 'message' in error) ? 
+            error.message : 
+            String(error)
+        )
+        .reduce((messages, error) => {
+          messages.push(error)
+          return messages
+        }, [])
+        .join(errorJoiner)
+    }`,
+    stack: errors.find(error => error.stack).stack
+  })
+}
+
+// Like Promise.all, but leaves no rejection/error unhandled by collecting them into an array
+//   and rejecting it at the end:
+export async function allPromisesHandled(promises = []){
+  // Ensure supplied list of promises is an array.
+  if (!Array.isArray(promises)) {
+    throw new TypeError('Supplied promises is not an Array')
+  }
+
+  // Array to collect promise errors
+  const errors = []
+
+  // Reduce all promises to a chain of promises resulting in an array of all promise results.
+  return await promises.reduce((chain, promise) => 
+    // With the chain of previous promise results,
+    chain.then(async results => {
```

Comment:
No need to async keyword here. Simply return `Promise.resolve` to chain

Reply:
Sounds good to me. I will refactor it in a new commit.